app/domain/authentication/authn_azure/validate_application_identity.rb

```diff
@@ -0,0 +1,105 @@
+require 'command_class'
+
+module Authentication
+  module AuthnAzure
+
+    Log = LogMessages::Authentication::AuthnAzure
+    Err = Errors::Authentication::AuthnAzure
+    # Possible Errors Raised: RoleNotFound, InvalidApplicationIdentity
+
+    ValidateApplicationIdentity = CommandClass.new(
+      dependencies: {
+        role_class: ::Role,
+        resource_class: ::Resource,
+        application_identity_class: ApplicationIdentity,
+        logger: Rails.logger
+      },
+      inputs: %i(account service_id username xms_mirid oid)
+    ) do
+
+      def call
+        token_identity_from_claims
+        # compare xms_mirid with what is defined in annotations
+        validate_application_identity
+      end
+
+      private
+
+      def application_identity
+        @application_identity ||= @application_identity_class.new(
+          role_annotations: role.annotations,
+          service_id: @service_id
+        )
+      end
+
+      def role_id
+        @role_id ||= @role_class.roleid_from_username(@account, @username)
+      end
+
+      def role
+        @role ||= @resource_class[role_id].tap do |role|
+          raise SecurityErr::RoleNotFound(role_id) unless role
+        end
+      end
+
+      # xms_mirid is a term in Azure to define a claim that describes the resource that holds the encoding of the instance's
+      # among other details the subscription_id, resource group, and provider identity needed for authorization.
+      # xms_mirid is one of the fields in the JWT token. This function will extract the relevant information from
+      # xms_mirid claim and populate a representative hash with the appropriate fields.
+      def token_identity_from_claims
+
+        # validates format of claim
```

Comment:
remove this comment. It is clear from the method name. Also remove the empty line above it.